resources/function-controller/charts/docker-registry/templates/deployment.yaml

```diff
@@ -71,19 +86,16 @@ spec:
           resources:
 {{ toYaml .Values.resources | indent 12 }}
           env:
-{{- if .Values.secrets.htpasswd }}
             - name: REGISTRY_AUTH
               value: "htpasswd"
             - name: REGISTRY_AUTH_HTPASSWD_REALM
               value: "Registry Realm"
             - name: REGISTRY_AUTH_HTPASSWD_PATH
-              value: "/auth/htpasswd"
-{{- end }}
+              value: "/data/htpasswd"
             - name: REGISTRY_HTTP_SECRET
-              valueFrom:
-                secretKeyRef:
-                  name: {{ template "docker-registry.fullname" . }}-secret
-                  key: haSharedSecret
+            # https://docs.docker.com/registry/configuration/#http, there's no problem that it is plainly seen 
```

Comment:
I am not convinced of this changes, because as I wrote, from my side, changing not out charts definitions is not a good practice - bumping charts in future will be a problematic (we will have a conflicts), but I know also a helm restrictions... At the moment it can stay that way